src/test/kotlin/deadline/application/DeadlineChangerTest.kt

```diff
@@ -0,0 +1,63 @@
+package deadline.application
+
+import deadline.domain.DeadlineRepository
+import deadline.domain.exceptions.AlreadyConfiguredDeadline
+import deadline.domain.exceptions.NonExistentDeadlineException
+import deadline.mothers.DeadlineMother
+import deadline.mothers.TimeMother
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.Mockito
+import shared.domain.exceptions.InvalidUUIDException
+import shared.mothers.IdentifierMother
+import kotlin.test.assertEquals
+
+class DeadlineChangerTest {
+
+    private lateinit var repository: DeadlineRepository
+    private lateinit var useCase: DeadlineChanger
+
+    @BeforeEach
+    fun `Set Up`() {
+        repository = Mockito.mock(DeadlineRepository::class.java)
+        useCase = DeadlineChanger(repository)
+    }
+
+    @Test
+    fun `Changing a deadline that doesn't exist throws NonExistingDeadlineException`() {
+        val deadline = DeadlineMother.getValidDeadline()
+        Mockito.`when`(repository.search(DeadlineMother.getNoteIdentifierFromDeadline(deadline))).thenReturn(null)
+        assertThrows<NonExistentDeadlineException> { useCase.change(deadline.toPrimitives()) }
+    }
+
+    @Test
+    fun `Getting an identifier that isn't valid throws InvalidUUIDException`() {
+        val x = DeadlinePrimitives(IdentifierMother.invalidPrimitiveIdentifier, TimeMother.getValidTime().time)
```

Comment:
Do not call variables like that, where have you been?
![](https://media.tenor.com/xJAFqppAJt8AAAAC/no-heck-no.gif)

Reply:
I've been doing your android 2 assigment, sorry about that, I will change it asap